library/common/jni/jni_interface.cc

```diff
@@ -362,6 +363,122 @@ static envoy_filter_trailers_status jvm_http_filter_on_response_trailers(envoy_h
                                         /*trailers*/ native_headers};
 }
 
+static void jvm_http_filter_set_request_callbacks(envoy_http_filter_callbacks callbacks,
+                                                  const void* context) {
+
+  __android_log_write(ANDROID_LOG_VERBOSE, "[Envoy]", "jvm_http_filter_set_request_callbacks");
+
+  JNIEnv* env = get_env();
+  jobject j_context = static_cast<jobject>(const_cast<void*>(context));
```

Comment:
I might have my rules mixed up but do we need to delete this local ref here?

Reply:
Yep, great catch. In a separate branch I started on last week, I'm adding caching to a bunch of the JNI refs, which in addition to being a noteworthy performance improvement, also reduces the number of locations we need to track refs like this and risk leaking.